core/terminal.go

```diff
@@ -102,79 +111,80 @@ func (container *Container) runTerminal(
 		return fmt.Errorf("failed to create service for interactive terminal: %w", err)
 	}
 
-	eg, egctx := errgroup.WithContext(ctx)
-
-	// forward a io.Reader to websocket
-	forwardFD := func(prefix []byte) func(r io.Reader) {
-		return func(r io.Reader) {
-			eg.Go(func() error {
-				for {
-					b := make([]byte, 512)
-					n, err := r.Read(b)
-					if err != nil {
-						if err == io.EOF {
-							return nil
-						}
-						return err
-					}
-					message := append([]byte{}, prefix...)
-					message = append(message, b[:n]...)
-					err = conn.WriteMessage(websocket.BinaryMessage, message)
-					if err != nil {
-						return err
-					}
-				}
-			})
-		}
+	bk := container.Query.Buildkit
+	term, err := bk.OpenTerminal(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to open terminal: %w", err)
 	}
 
+	fmt.Fprintf(term.Stderr, "Attaching interactive terminal: ")
+	idtui.DumpID(idtui.NewOutput(term.Stderr, termenv.WithTTY(true)), svcID)
+	fmt.Fprintf(term.Stderr, "\r\n\n")
+	eg, egctx := errgroup.WithContext(ctx)
 	runningSvc, err := svc.Start(
 		ctx,
 		svcID,
 		true,
-		func(w io.Writer, svcProc bkgw.ContainerProcess) {
+		func(stdin io.Writer, svcProc bkgw.ContainerProcess) {
+			eg.Go(func() error {
+				_, err := io.Copy(stdin, term.Stdin)
+				if err != nil {
```

Comment:
should this get the same "ignore `io.ErrClosedPipe`" check as the other copies below?